include/tnt/linear/orthogonal_projection.hpp

```diff
@@ -0,0 +1,46 @@
+#ifndef TNT_ORTHOGONAL_PROJECTION_HPP
+#define TNT_ORTHOGONAL_PROJECTION_HPP
+
+#include <tnt/core/tensor.hpp>
+#include <type_traits>
+
+namespace tnt
+{
+
+    namespace detail
+    {
+
+        template <typename DataType, typename Enable = void>
+        struct OptimizedOrthogonalProjection
+        {
+            static Tensor<DataType> eval(const Tensor<DataType>&, const Tensor<DataType>&);
```

Comment:
The inputs are `const` here but not in your *_impl.hpp file. I think they should be const in both places